### ISSUE TITLE
[dev.icinga.com #8439] downtime configuration / execution

### ISSUE THREAD

**Issue by icinga-migration** [Icinga/icinga2#2619]
This issue has been migrated from Redmine: https://dev.icinga.com/issues/8439

**Created by Tami on 2015-02-14 10:56:56 +00:00**

Assignee: _(none)_
Status: _Rejected_  (closed on _2015-03-19 10:07:43 +00:00_)
Target Version: _(none)_
Last Update: _2015-03-19 10:07:43 +00:00 (in Redmine)_

    Icinga Version: r2.2.4-1
    Backport?: Not yet backported
    Include in Changelog: 1

---

Hello,

i configured a Nightly Downtime which should happen on every weekday (Mo-Fr) starting 22:30 running up to 06:00 of the next day.

downtimes.conf:
> apply ScheduledDowntime "Nightly-Backup" to Host {
> author = "itsupport"
> comment = "Geplante downtime fuer Backup"
> 
> ranges = {
> monday = host.vars.bup\_nightly\_downtime
> tuesday = host.vars.bup\_nightly\_downtime
> wednesday = host.vars.bup\_nightly\_downtime
> thursday = host.vars.bup\_nightly\_downtime
> friday = host.vars.bup\_nightly\_downtime
> }
> assign where host.vars.bup\_nightly\_downtime != ""
> }

The hosts that shut down at 22:30 for the backup:
> object Host "DC-WBSZBB" {
> import "dc-template"
> address = "10.120.62.42"
> vars.os = "Domaincontroller"
> vars.bup\_nightly\_downtime = "22:30-24:00,00:00-06:00"
> }
> 
> object Host "DC-WBSA" {
> import "dc-template"
> address = "10.120.70.42"
> vars.os = "Domaincontroller"
> vars.bup\_nightly\_downtime = "22:30-30:00"
> }

And this is what i see in Icinga Web 2:
![icingaweb2_downtimes.JPG](https://dev.icinga.com/attachments/download/2264/icingaweb2_downtimes.JPG)

For me the confusing part is, that unless i configure the downtime as 22:30-30:00 it won't show up as 22:30 to 06:00 of the next day as downtime. (DC-WBSA)
The configuration of DC-WBSZBB as it is will only show the downtime from 22:30 to 24:00.
(This was before the downtime actually happened last night, the screenshot above is now that the downtimes are over.)

Going by the E-Mail notifcations i got the DC-WBSA 22:30-30:00 downtime seems to be the one working correctly, but only expires on the tuesday and i don't have any idea why.
(We don't have any downtimes on Saturday and Sunday which is why i left those days out of the configuration of downtimes.conf)

The DC-WBSZBB downtime ended at 24:00 going by the email notification.

DC-WBSA downtime email notification:
> ****\* Icinga ****\*
> 
> Notification Type: DOWNTIMESTART
> 
> Host: DC-WBSA
> Address: XXX.XXX.XXX.XXX
> State: DOWN
> 
> Date/Time: 2015-02-13 22:33:27 +0100
> 
> Additional Info: PING CRITICAL - Packet loss = 100%
> 
> Comment: \[\]

> ****\* Icinga ****\*
> 
> Notification Type: DOWNTIMEEND
> 
> Host: DC-WBSA
> Address: XXX.XXX.XXX.XXX
> State: UP
> 
> Date/Time: 2015-02-14 06:03:02 +0100
> 
> Additional Info: PING OK - Packet loss = 0%, RTA = 24.85 ms
> 
> Comment: \[\]

DC-WBSZBB downtime email notification:
> ****\* Icinga ****\*
> 
> Notification Type: DOWNTIMESTART
> 
> Host: DC-WBSZBB
> Address: XXX.XXX.XXX.XXX
> State: UP
> 
> Date/Time: 2015-02-13 22:30:02 +0100
> 
> Additional Info: PING OK - Packet loss = 0%, RTA = 23.83 ms
> 
> Comment: \[\]

> ****\* Icinga ****\*
> 
> Notification Type: DOWNTIMEEND
> 
> Host: DC-WBSZBB
> Address: XXX.XXX.XXX.XXX
> State: DOWN
> 
> Date/Time: 2015-02-14 00:00:28 +0100
> 
> Additional Info: PING CRITICAL - Packet loss = 100%
> 
> Comment: \[\]
(don't be confused by the state of DC-WBSZBB, the shutdown of the machines starts somewhat later as indicated properly on DC-WBSA which went down at 22:33, but the backup script starts at 22:30)
As you can see, the notification for DC-WBSZBB for DOWNTIMEEND came at 00:00 while the backup process was still running and the machine still offline.
As for DC-WBSA it worked properly, but the downtime expiration date is off the scale.

So either i configured something totaly wrong or there is a bug.
(And i havent found anything in the latest documentation yet which would tell me to do anything differently unless STRG-F failed me.)
Considering there was an issue with the downtime before where a configuration "22:00-09:00" would cause a segfault, which was fixed and i was told my logic was just fine i was hoping "22:30-06:00" should work now, though to me it seems it does not work either way. :(

Not entirely sure if this is a Icinga2 or Icinga Web 2 bug, but going by the notifications which have nothing to do with the Web part as far i can tell... i would guess it's within Icinga2.

Icinga Version: r2.2.4-1
Icinga Web 2: cloned from Git on Wednesday
Linux Distribution: Debian 7.8 (wheezy)

If any more information is required, please let me know.
I also frequently visit IRC ( Tami\_ ).

Disclaimer:
Please forgive my poor skills when it comes down to coding and tracking bugs, i don't mind light slaps on the back of my head either in case i did something wrong. ;)
I haven't found any related downtime issue that would still be open by the Search on the site either.

**Attachments**

* [icingaweb2_downtimes.JPG](https://dev.icinga.com/attachments/download/2264/icingaweb2_downtimes.JPG) Tami - _2015-02-14 10:28:24 +00:00_
